lib/streamlit/elements/select_slider.py

```diff
@@ -0,0 +1,107 @@
+from streamlit.proto.Slider_pb2 import Slider as SliderProto
+from streamlit.errors import StreamlitAPIException
+from streamlit.type_util import ensure_iterable
+from .utils import _get_widget_ui_value
+
+
+class SelectSliderMixin:
+    def select_slider(
+        dg, label, options=[], value=None, format_func=str, key=None,
+    ):
+        """Display a slider widget on a discrete set of options of any type.
```

Comment:
How about:

```
"""
Display a slider widget to select items from a list.

This also allows you to render a range slider by passing a two-element tuple or list as the `value`.

The difference between `st.select_slider` and `st.slider` is that `select_slider` accepts any datatype and takes a list as input, while `slider` only accepts numerical data and takes a range as input.
```

Then also add that last paragraph to the `st.slider` docs.

Reply:
I can do that. Numerical is misleading though since `slider` accepts dates right?

Reply:
I'll just used `only accepts numerical or date/time data`